module/scratchpad.lua

```diff
@@ -163,17 +163,28 @@ end
 --- Turns the scratchpad off if it is focused otherwise it raises the scratchpad
 function Scratchpad:toggle()
     local is_turn_off = false
+    local matches = self:find()
     if self.dont_focus_before_close then
-        local matches = self:find()
-        if matches[1] and matches[1].first_tag then
-            is_turn_off = matches[1].first_tag.selected
+        if matches[1] then
+            local current_tag = matches[1].screen.selected_tag
+            for k, tag in pairs(matches[1]:tags()) do
+                if tag == current_tag then
+                    is_turn_off = true
+                    break
+                else
+                    is_turn_off = false
+                end
+            end
         end
     else
         is_turn_off = client.focus and
                           awful.rules.match(client.focus, self.rule)
     end
 
     if is_turn_off then
+        if self.disable_floating_on_close and matches[1].floating == true then
```

Comment:
When turning on a scratchpad it will make the client floating, so I think the default sane option would be to make the client not floating back again when closing (unless it was floating before)

Reply:
I thought it makes sense to include it as an option as when turning on the scratchpad it will make the client floating. If I'm being honest, I think the default option should be true so it automically makes the client not floating again once closing.

Reply:
If you want your scratchpad to not float then you can just set `floating` to false (and maybe set `reapply` to true) in the options. I'm a little confused in what case that extra option is needed to be honest. You are on board with me that signal support would make that option obsolete though, right?

Reply:
@Nooo37 I do want it to float when the scratchpad is open, but when I close the scratchpad it should go back to non floating. Does it not make sense? Yeah, the signal support would make this obsolete, but wouldn't it be more sane to set the client to non floating after closing the scratchpad?

Reply:
I don't quite understand. Why do you care whether your scratchpad is floating or not when it is closed?

Reply:
@Nooo37 Well, the closed scratchpad might still be opened in another tag